src/GGTestUtils.py

```diff
@@ -300,6 +349,86 @@ def wait_for_deployment_till_timeout(
                                             "coreDeviceExecutionStatus"]) ==
                                           "FAILED"):
+                                        print(f"\n{'='*60}")
+                                        print(
+                                            f"DEPLOYMENT FAILED: {deployment_id}"
+                                        )
+                                        print(f"Thing: {thing}")
+                                        print(
+                                            f"Status Reason: {deployment.get('statusReason', 'N/A')}"
+                                        )
+                                        print(
+                                            f"Full deployment details: {deployment}"
+                                        )
+
+                                        # Get full deployment details from AWS
+                                        try:
+                                            aws_deployment = self._ggClient.get_deployment(
+                                                deploymentId=deployment_id)
+                                            print(
+                                                f"\nAWS Deployment Status: {aws_deployment.get('deploymentStatus', 'N/A')}"
+                                            )
+                                            print(
+                                                f"AWS Deployment Policies: {aws_deployment.get('deploymentPolicies', {})}"
+                                            )
+                                            if 'components' in aws_deployment:
+                                                print(
+                                                    f"Components in deployment: {list(aws_deployment['components'].keys())}"
+                                                )
+                                        except Exception as e:
+                                            print(
+                                                f"Could not get AWS deployment details: {e}"
+                                            )
+
+                                        print(
+                                            f"\nChecking all Greengrass logs for errors..."
+                                        )
+                                        try:
+                                            import subprocess
+                                            # Check all greengrass services
+                                            services = [
+                                                "ggdeploymentd", "iotcored",
+                                                "ggconfigd", "tesd",
+                                                "gghealthd", "ggipcd"
+                                            ]
+                                            for svc in services:
+                                                log_output = subprocess.run(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Starting a process with a partial executable path https://bandit.readthedocs.io/en/latest/plugins/b607_start_process_with_partial_path.html